### PR TITLE
Initial implementation of laswp.

### PR DESCRIPTION
**Pull request by rleonid**
@mmottl There were a couple of things that I wasn't certain about. For example 

> caml_enter_blocking_section();  /* Allow other threads */

I thought that entering a blocking section was about preventing other threads from running, not allowing?

### CODE REVIEW

File: lib/lacaml_utils.ml [hunk 1]
```diff
@@ -147,6 +149,14 @@ external ilaenv : int -> string -> string -> int -> int -> int -> int -> int
 let check_var_ltz loc var_name var =
   if var < 0 then invalid_arg (sprintf "%s: %s < 0: %d" loc var_name var)
 
+let check_var_within loc var_name var lb ub c =
```
Comment (rleonid) [inline lib/lacaml_utils.ml]: Also not certain if this function belongs here, but I figured since I use it twice to check variables, it seemed natural.

**Comment by mmottl** [comment]
@rleonid Indeed, the name of the `caml_enter_blocking_section` function is very unfortunate.  Here a "blocking section" blocks on e.g. some system call (or at least takes a long time), it's not about blocking entry to the runtime.  INRIA realized that, obsoleted the old name, and now has functions `caml_release_runtime_system` and `caml_acquire_runtime_system`.  Feel free to use those in new code.

You can keep putting new checking functions in `lacaml_utils.ml`.

I have to say, both the manual of the `xLASWP` function and its Fortran API are godawful.  I read the Fortran code to understand it better, and I think it's inconsistent with what the manual says.  The API is also totally counter-intuitive.  `K1` and `K2` surely don't indicate anything the user might expect.  I would bet the author did not intend this semantics.  Probably the only reason why people haven't noticed so far is, because almost surely nobody uses increments other than `1` and `-1`.

This is a problem, because we need to implement the correct checks in LACAML to prevent segfaults.  I have filed a bug report to LAPACK at http://icl.cs.utk.edu/lapack-forum/viewtopic.php?f=13&t=4798.  I hope they will let us know soon what the intended behavior should be.  I'll check out the pull request then.

**Comment by rleonid** [comment]
I've update the PR to use `caml_release_runtime_system`.

Glad to hear that I wasn't the only one confused by `incx.` At first I interpreted it to mean as how the counter inside of`ipiv` is incremented, but it seems to mean that row `r` is swapped with `ipiv(k1+incx*r)`

This isn't urgent and can wait, probably a `` `F ``(orward) or `` `B ``(ackward) interface makes more sense.

**Comment by rleonid** [comment]
I'm seeing some weird behavior with `forward=false` on `lapmt` so the latest commit isn't foolproof. I'll do some more testing to diagnose the problem.

**Pull request merged**